ed25519/src/lib.rs

```diff
@@ -254,10 +254,16 @@
 #![forbid(unsafe_code)]
 #![warn(missing_docs, rust_2018_idioms, unused_qualifications)]
 
+#[cfg(all(
+    feature = "serde_bytes",
+    any(not(feature = "serde"), not(feature = "std"))
+))]
+compile_error!("Feature \"serde_bytes\" requires \"std\" and \"serde\" features to be enabled");
```

Comment:
It seems like it should be possible to make this "just work", although if you'd prefer I can tackle that as a followup.

Reply:
I did spend some time playing with it, but wasn't able to make it work. I felt blocked by the limitation against having features in `Cargo.toml` with the same name as an optional dependency. I did use a separate feature at one point, like:

```toml
[features]
serde_bytes_support = ["serde", "serde_bytes", "std"]
```

This works, but I felt it wasn't much better since it doesn't remove the implicit `serde_bytes` feature from the optional dependency (so you could still compile with just `--features serde_bytes` and get either missing type errors or silent nothing).

Reply:
You can prevent a clash between the crate name and the feature name by renaming the crate on import:

```toml
serde_bytes_crate = { package = "serde_bytes", version = "0.11", optional = true }
```

Reply:
Cool, that would allow for something like:

```toml
[features]
serde_bytes = ["serde", "serde_bytes_crate", "std"]
```

I suppose that would leave an implicit `serde_bytes_crate` feature, which is also a bit annoying, but if it's preferable to the `compile_error!` I can make that change.

Reply:
That can eventually be sorted out / cleaned up when namespaced features land